howso/engine/trainee.py

```diff
@@ -1307,26 +1250,26 @@ def predict(
 
     def react(
         self,
-        contexts: Optional[Union[List[List[object]], "DataFrame"]] = None,
+        contexts: Optional[DataFrame | List[List[object]]] = None,
```

Comment:
This is used a lot, perhaps we could define a type `TabularData = DataFrame | List[List[Any]]` and use that everywhere?